File: 3'rd Semester/DSA/leetcode/build_array_from_permutation.c
#include <stdio.h>
#include<stdlib.h>

int main(int argc, char const *argv[])
{
    int nums[] = {0, 2, 1, 5, 3, 4};
    int numsSize = 6;
    int *ans = (int *)malloc(numsSize * sizeof(int));

    for (int i = 0; i < numsSize; i++)
    {
        ans[i] = nums[nums[i]];
    }

    return ans;
}

File: 3'rd Semester/DSA/leetcode/build_array_from_permutation.java
class build_array_from_permutation {
    public static void main(String[] args) {
        int nums[] = {0,2,1,5,3,4};
        int ans[] = new int[nums.length];

        for (int i = 0; i < ans.length; i++) {
            ans[i] = nums[nums[i]];
        }

        for (int i : ans) {
            System.out.println(i);
        }
    }
}